src/npg_notify/porch_wrapper/qc_state.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+import os
+from urllib.parse import urljoin
+
+import requests
+from npg_notify.config import get_config_data
+from npg_porch_cli.api import Pipeline, PorchAction
+from npg_porch_cli.api import send as send_porch_request
+
+LANGQC_CONFIG_FILE_SECTION = "LANGQC"
+PORCH_CONFIG_FILE_SECTION = "PORCH"
+
+VERIFY_CERTIFICATE = False  # TODO: change to True
+CLIENT_TIMEOUT = (10, 60)
+
+
+def run():
+    parser = argparse.ArgumentParser(
+        prog="qc_state_notification",
+        description="Creates or processes notifications about QC states.",
+    )
+    parser.add_argument(
+        "action",
+        type=str,
+        help="Action to send to npg_porch server API",
+        choices=["register"],
+    )
+    parser.add_argument(
+        "--conf_file_path",
+        type=str,
+        required=True,
+        help="Configuration file path.",
+    )
+    args = parser.parse_args()
+
+    create_tasks(conf_file_path=args.conf_file_path)
+
+
+def create_tasks(conf_file_path: str):
+    """Retrieves and registers with Porch recently assigned QC states.
+
+    Retrieves from LangQC API all recently (within the last four weeks) assigned
+    final QC states for PacBio wells. Registers these states as pending with
+    one of the pipelines registered with Porch API. URLs to use with these APIs,
+    details of the Porch pipeline and Porch authorization token should be given
+    in the configuration file.
+
+    All errors in task registration are captured and logged.
+
+    Args:
+      conf_file_path:
+        Configuration file path for this application.
+    """
+
+    porch_conf = get_config_data(
+        conf_file_path=conf_file_path,
+        conf_file_section=PORCH_CONFIG_FILE_SECTION,
+    )
+    # query LangQC for recently QC-ed wells
+    pipeline = Pipeline(
+        name=porch_conf["pipeline_name"],
+        uri=porch_conf["pipeline_uri"],
+        version=porch_conf["pipeline_version"],
+    )
+
+    langqc_conf = get_config_data(
+        conf_file_path=conf_file_path,
+        conf_file_section=LANGQC_CONFIG_FILE_SECTION,
+    )
+    qc_states = _send_request(
+        url=urljoin(langqc_conf["api_url"], langqc_conf["recently_qced_url"])
+    )
+
+    num_products = len(qc_states)
+    print(
+        f"Retrieved QC states for {num_products} products."
+    )  # TODO: log properly
+
+    os.environ["NPG_PORCH_TOKEN"] = porch_conf["npg_porch_token"]
+    del porch_conf["npg_porch_token"]
+
+    # qc_states is a dictionary where keys are product IDs and values are
+    # lists of QC states. A list of one QC state is expected.
```

Comment:
The list of one is sufficiently surprising that I think that we could do with a little more exposition:

"... one expected because we are limiting to sequencing products only"